### ISSUE TITLE
Advancing braille should dismiss alerts

### ISSUE THREAD

**Issue by mehgcap** [nvaccess/nvda#15697]
### Is your feature request related to a problem? Please describe.
No.

### Describe the solution you'd like
When an alert is shown in braille, panning the display past the end of the alert text should dismiss the alert and show what the display would have shown had the alert not been there. Put more simply, NVDA should treat alert text like part of the text being shown. Pan past the alert, and the next refresh shows the text the alert replaced when it appeared.

One example I run into a lot is VSCode. If I press ctrl-p to do a search of the code base for files, the list of files gets filtered as I type. The problem is that NVDA shows an alert as well. Instead of the display showing me the top result, the alert showing how many files were matched covers it up. Instead of my display showing "myfile.txt" as I start to type "myfile", the display instead shows "11 results".

Currently, if I pan the display while "11 results" is showing, nothing happens. My proposal is that panning beyond this line of text should dismiss the alert and show me the top result in the list, just as would have been shown had the alert not been displayed at all.

I can easily imagine that this long-standing behavior suddenly changing would be confusing and disruptive to many people. It should be a toggle in settings, disabled by default. Only those who want to turn on the new alert dismissal method will do so. For everyone else, things stay the same.

### Describe alternatives you've considered
Not really applicable.

### Additional context
None I can think of.

**Comment by CyrilleB79** [comment]
Does pressing routing cursor dismiss the alert?

**Comment by mehgcap** [comment]
Yes, a cursor routing key dismisses the alert. I was unaware of this, so it's nice to have. I don't think this fully addresses my suggestion, though, as it's nice to simply scroll an alert out of the way instead of taking a hand off the display to hit the cursor key. I use a Humanware display with thumb keys, so I don't have to reach over to a rocker or button like on other displays.
